shared/structs.go

```diff
@@ -23,3 +40,45 @@ func (s *Scan) AddHashed(path string, hash uint32) {
 func (s *Scan) AddErrored(path string, err error) {
 	s.Files = append(s.Files, File{path, err, 0})
 }
+
+func (s *Scan) Diff(other *Scan) *Diff {
+	diff := Diff{nil, nil, []DiffFile{}}
+	diff.AddName(s.Plugin, other.Plugin)
+	diff.AddVersion(s.Version, other.Version)
+
+	for _, otherFile := range other.Files {
+		found := false
+
+		for _, file := range s.Files {
```

Comment:
I'm not wild about this `O(n^2)` comparison and would like to speed it up if it becomes a problem., but I figured that KISS is important in the beginning.